src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -24,14 +24,23 @@ public LinkAttributesStep (XPathDocument document, EmbeddedResource resource, As
 		{
 		}
 
-		IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav)
+		IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttributeProvider provider = null)
 		{
 			XPathNodeIterator iterator = nav.SelectChildren ("attribute", string.Empty);
 			var attributes = new List<CustomAttribute> ();
 			while (iterator.MoveNext ()) {
+				if (!ShouldProcessElement (iterator.Current))
+					continue;
+
 				AssemblyDefinition assembly;
 				TypeDefinition attributeType;
 
+				string internalAttribute = GetAttribute (iterator.Current, "internal");
+				if (internalAttribute == "RemoveAttributeInstances" && provider != null) {
+					Context.Annotations.AddRemovableAttribute ((IMemberDefinition) provider);
```

Comment:
This won't work correctly. Basically if this type has more than one attribute from the XML (or from multiple XMLs) it will lose all but the remove one. Calling this method on the annotations will do the "cache the attributes" in the linker, but at that point the CustomAttributes is not yet populated with all XML attributes - so it will only see some attributes, not all.

Reply:
I would probably solve it like this:
* add a new Dictionary to CustomAttributeSource - something like _internalAttributes which will be `Dictionary<ICustomAttributeProvider, IEnumerable<Attribute>>`
* add `AddInternalAttribute`, `HasInternalAttributes` and `GetInternalAttributes` to the class
* teach the `LinkerAttributesInformation` about the second set of internal attributes so that it populates the cache from those as well.

Reply:
Add a test for this (basically the RemoveAttributeInstances followed by some test attribute on the same type, both should take effect.